full/src/main/java/apoc/uuid/UUIDHandlerNewProcedures.java

```diff
@@ -0,0 +1,118 @@
+        final UuidInfo[] result = new UuidInfo[1];
+
+        withSystemDb(sysTx -> {
+            Node node = Util.mergeNode(sysTx, SystemLabels.ApocUuid, null,
+                    Pair.of(database.name(), databaseName),
+                    Pair.of(SystemPropertyKeys.label.name(), label)
+            );
+
+            node.setProperty(propertyName.name(), config.getUuidProperty());
+            node.setProperty(addToSetLabel.name(), config.isAddToSetLabels());
+            node.setProperty(addToExistingNodes.name(), config.isAddToExistingNodes());
+
+            // we'll the return current uuid info
+            result[0] = new UuidInfo(node, true);
+
+            setLastUpdate(sysTx, databaseName, ApocUuidMeta);
+        });
+
+        return result[0];
+    }
+
+    public static UuidInfo drop(String databaseName, String labelName) {
+        final UuidInfo[] previous = new UuidInfo[1];
+
+        withSystemDb(tx -> {
+            getUuidNodes(tx, databaseName, Map.of(SystemPropertyKeys.label.name(), labelName))
+                    .forEachRemaining(node -> {
+                        previous[0] = new UuidInfo(node);
+                        node.delete();
+                    });
+
+            setLastUpdate(tx, databaseName, ApocUuidMeta);
+        });
+
+        return previous[0];
+    }
+
+    public static List<UuidInfo> dropAll(String databaseName) {
+        final List<UuidInfo> previous = new ArrayList<>();
+
+        withSystemDb(tx -> {
+            getUuidNodes(tx, databaseName)
+                    .forEachRemaining(node -> {
+                        // we'll return previous uuid info
+                        previous.add( new UuidInfo(node) );
+                        node.delete();
+                    });
+
+            setLastUpdate(tx, databaseName, ApocUuidMeta);
+        });
+
+        return previous;
+    }
+
+    public static ResourceIterator<Node> getUuidNodes(Transaction tx, String databaseName) {
+        return getUuidNodes(tx, databaseName, null);
+    }
+
+    public static ResourceIterator<Node> getUuidNodes(Transaction tx, String databaseName, Map<String, Object> props) {
+        return getSystemNodes(tx, databaseName, SystemLabels.ApocUuid, props);
+    }
+
+    public static void checkConstraintUuid(Transaction tx, String label, String propertyName) {
+        Schema schema = tx.schema();
+        Stream<ConstraintDefinition> constraintDefinitionStream = StreamSupport.stream(schema.getConstraints(Label.label(label)).spliterator(), false);
+        boolean exists = constraintDefinitionStream.anyMatch(constraint -> {
+            Stream<String> streamPropertyKeys = StreamSupport.stream(constraint.getPropertyKeys().spliterator(), false);
+            return streamPropertyKeys.anyMatch(property -> property.equals(propertyName));
+        });
+        if (!exists) {
+            String error = String.format("`CREATE CONSTRAINT ON (%s:%s) ASSERT %s.%s IS UNIQUE`",
```

Comment:
new syntax perhaps even with a constraint name and IF NOT EXISTS

Reply:
Added a test to check it (`testConstraintAutoCreation`)